pkg/controller/jobframework/interface.go

```diff
@@ -62,13 +62,13 @@ type JobWithReclaimablePods interface {
 	ReclaimablePods() ([]kueue.ReclaimablePod, error)
 }
 
-type StopReason int
+type StopReason string
 
 const (
-	StopReasonWorkloadDeleted StopReason = iota
-	StopReasonWorkloadEvicted
-	StopReasonNoMatchingWorkload
-	StopReasonNotAdmitted
+	StopReasonWorkloadDeleted    StopReason = "WorkloadDeleted"
+	StopReasonWorkloadEvicted    StopReason = "WorkloadEvicted"
```

Comment:
Uhm... this one might not be enough. We also need to know if it was WaitForPodsReady or Preemption

Reply:
Ok, a second try. Please take a look. I don't know if I should add to this enum all possible Eviction reasons or I can just directly cast and propagate the `evCond.Reason` reason?

Reply:
As discussed offline, let's see if we can use a hyphen to concatenate the reasons.

Reply:
@alculquicondor Could we decide the delimiter based on this discussion https://github.com/kubernetes/enhancements/pull/4479/files#r1581226166 since the reason should be CamelCase?

> In condition types, and everywhere else they appear in the API, Reason is intended to be a one-word, CamelCase representation of the category of cause of the current status

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties

That discussion will seem to happen at the next sig apps meeting.

Reply:
> That discussion will seem to happen at the next sig apps meeting.

Oh, there is not the discussion in the agenda...

Reply:
I'm just trying to make the reason make sense.

`WorkloadEvictedPreempted` doesn't read that well.

So, the alternatives would be:
1. `WorkloadPreempted` (drop the `Evicted` part)
2. `Preempted` (just use the reason directly, and forget about the Workload bit)
3. `WorkloadEvictedBecausePreempted` (same proposal as the hyphen, but with a word)

Maybe 1 reads best of all? WDYT?

Reply:
I agree with you since we have already the `Evicted` in the `condition.type`, right?

Reply:
So the possible reasons for eviction are: https://github.com/kubernetes-sigs/kueue/blob/b870144dc272788a755369411052bf5b81e50348/apis/kueue/v1beta1/workload_types.go#L332-L351

And 1. reads definitely good for "Preempted" but for the other ones it would be a bit confusing. In my opinion concatenation with the dash would make the most sense. But if it is not recommended in the guidelines then we can decide on any of the alternatives. Would `WorkloadEvictedWithReason{Preempted,PodsReadyTimeout ...}` (a variant of 3) be too long?

Reply:
I like it, even if it's too long

Reply:
Ok, changed the joining word to "Reason".